### ISSUE TITLE
Unicode module

### ISSUE THREAD

**Issue by ivankokan** [vectorgraphics/asymptote#142]
The implementation of `unicode` module is out-of-date for some time. Although `ucs` package served its purpose years ago (along with `utf8x` option for `inputenc`), it has been always suggested to avoid using it: https://tex.stackexchange.com/a/13070/115879.

Nowadays, such implementation is not needed at all:
https://tex.stackexchange.com/a/184628/115879
https://tex.stackexchange.com/a/203804/115879.

In other words, both `ucs` package and `utf8x` option for `inputenc` should not be used anymore (nevertheless, `utf8` has been the default one since 2018: https://tex.stackexchange.com/questions/446650/please-use-latin2-or-utf8-inputenc-with-magyar-babel-warning/446653#446653).

Proposal: let's completely remove this module, I can prepare the PR.

**Comment by jorgepiloto** [comment]
Hi @ivankokan, 

I recently started using asymptote and wanted to generate a label with zodiac unicode characters, i.e.: [pisces symbol](https://www.fileformat.info/info/unicode/char/2653/index.htm). After #144 was merged, the `unicode` module is no longer available (and thus required) and thought this should be as simple as running:

```c
# Generate a label showing pisces symbol
label("\u2648", (0,0));
```

However, I am not getting desired output and just the numbers are printed. This is probable simple to solve but even after reading documentation, I wasn't able to achieve desired output. May you please guide or notice me if is a local problem or am I doing something wrong? Thank you for this amazing tool!

**Comment by ivankokan** [comment]
Please check this FAQ: https://asymptote.sourceforge.io/FAQ/section4.html#internatfonts

I am in a hurry, will try to test and answer properly later today.

**Comment by charlesstaats** [comment]
This may or may not work, but try changing your double-quotes " to single
quotes '. That way your backslash will be processed as an escape character
rather than forwarded to TeX as a backslash.

**Comment by jorgepiloto** [comment]
Thank you both @ivankokan  and @charlesstaats for your quick answers on this. Although I was not lucky with those suggestions, I ended up using the following LaTeX packages for writing down symbols in labels: `wasysym` and `marvosym`. However, I would still like to know how to face this problem without importing LaTeX packages, although now this issue is not critical for me. Best regards :rocket:

**Comment by ivankokan** [comment]
@jorgepiloto Can you please provide the original example you had troubles with?

**Comment by jorgepiloto** [comment]
Thanks for your help @ivankokan, let me paste a reproducible script which contains also additional information regarding my Asymptote version and Latex one:

```asymptote
/*
 * Reproducible bug when using LaTeX labels

 * Asymptote version 2.67 [(C) 2004 Andy Hammerlindl, John C. Bowman, Tom Prince]
 * ------------------------------------------------------------------------------

 * ENABLED OPTIONS:
 * GSL      GNU Scientific Library (special functions)
 * FFTW3    Fast Fourier transforms
 * XDR      external data representation (portable binary file format)
 * CURL     URL support
 * Readline interactive history and editing
 * Sigsegv  distinguish stack overflows from segmentation faults
 * GC       Boehm garbage collector
 * 
 * DISABLED OPTIONS:
 * WebGL    3D HTML rendering
 * OpenGL   3D OpenGL rendering


 * LaTeX version
 * -------------

 * pdfTeX 3.14159265-2.6-1.40.21 (TeX Live 2020/Arch Linux)
 * kpathsea version 6.3.2
 * Copyright 2020 Han The Thanh (pdfTeX) et al.
 * There is NO warranty.  Redistribution of this software is
 * covered by the terms of both the pdfTeX copyright and
 * the Lesser GNU General Public License.
 * For more information about these matters, see the file
 * named COPYING and the pdfTeX source.
 * Primary author of pdfTeX: Han The Thanh (pdfTeX) et al.
 * Compiled with libpng 1.6.37; using libpng 1.6.37
 * Compiled with zlib 1.2.11; using zlib 1.2.11
 * Compiled with poppler version 0.90.0

 */

/* Packages and modules to be used */
import geometry;

/* Settings, configuration and packages */
settings.outformat="png";
unitsize(1cm);


/* Macros and constants */
pair OO = (0,0);
real R_circle = 2.50;


/* Draw a simple circle with pisces unicode symbol inside of it */
draw(circle(OO, R_circle), linewidth(2pt));
label("\u2648", OO);
```

**Comment by jorgepiloto** [comment]
Previous script generates the following figure without properly rendering unicode label. Not sure if this is the proper way to define these kind of characters...

<div align="center">
<img src="https://user-images.githubusercontent.com/28702884/93738894-38f7c300-fbe7-11ea-98c4-7f6b53be5a27.png">
</div>

**Comment by ivankokan** [comment]
**Has such code ever worked within Asymptote (before removing the `unicode` module)?**

Manually adding the equivalent of `unicode` does not bring any difference:
```asy
usepackage("ucs");
usepackage("inputenc","utf8x");
```

**Comment by jorgepiloto** [comment]
Most of my work makes use of Python language, which provides the syntax `"\uXXXX" in case you want to print unicode characters. I am a little bit familiar with C/C++ syntax and, since Asymptote shows a similar one, thought it might be also possible to generate unicode strings this way.

I recently started to use the Asymptote vector drawing language and never check if importing `unicode module made a difference in the behavior of the script. Also, I tried the following with both `ucs` and `inputenc` modules without success:

```asymptote
label("♓︎", OO)
```

**Comment by jorgepiloto** [comment]
By the way, this is the image I was able to generate with the usage of `wasysym` and `marvosym` packages:

<div align="center">
<img height="400px" src="https://user-images.githubusercontent.com/28702884/93810703-519bc380-fc4f-11ea-8272-dfc86581f016.png">
</div>

<br>

Your software is amazing, thanks for all responses and helping me with this issue :rocket:

**Comment by ivankokan** [comment]
> Most of my work makes use of Python language, which provides the syntax `"\uXXXX" in case you want to print unicode characters. I am a little bit familiar with C/C++ syntax and, since Asymptote shows a similar one, thought it might be also possible to generate unicode strings this way.
> 
> I recently started to use the Asymptote vector drawing language and never check if importing `unicode` module made a difference in the behavior of the script. Also, I tried the following with both `ucs` and `inputenc` modules **without success**:
> 
> ```c++
> label("♓︎", OO)
> ```

Let us first conclude:
- Asymptote labels are generated by using some TeX engine in the background.
- `\u####`-syntax is not supported by any TeX engine (as far as I know).
- Removing the previously mentioned Asymptote modules did not make any damage.

Later today I will provide examples on how to create Unicode labels depending on some TeX engines I am familiar with.
